transport-native-epoll/src/main/java/io/netty/channel/epoll/EpollEventLoop.java

```diff
@@ -394,24 +397,25 @@ protected void run() {
                 if (ioRatio == 100) {
                     try {
                         if (strategy > 0 && processReady(events, strategy)) {
-                            prevDeadlineNanos = Long.MAX_VALUE;
+                            prevDeadlineNanos = NONE;
                         }
                     } finally {
                         // Ensure we always run tasks.
                         runAllTasks();
                     }
-                } else {
+                } else if (strategy > 0) {
                     final long ioStartTime = System.nanoTime();
-
                     try {
-                        if (strategy > 0 && processReady(events, strategy)) {
-                            prevDeadlineNanos = Long.MAX_VALUE;
+                        if (processReady(events, strategy)) {
+                            prevDeadlineNanos = NONE;
                         }
                     } finally {
                         // Ensure we always run tasks.
                         final long ioTime = System.nanoTime() - ioStartTime;
                         runAllTasks(ioTime * (100 - ioRatio) / ioRatio);
```

Comment:
this `(100 - ioRatio) / ioRatio` has to be called on each iteration? `/` seems quite heavy

Reply:
@franz1981 maybe also for a separate PR since it's always been like this AFAIK... though I think if we want to calculate an accurate percentage of the measured time then a div will have to be done at some point, unless we want do a floating point multiply (and conversion to/from fp).

Also we had been thinking of getting rid of this altogether, and always process all tasks (with some limit on number of outer iterations) - see https://github.com/netty/netty/pull/9590#issuecomment-542054426. After more thought I feel like that makes more sense actually... thoughts?